pkg/handlers/scan_test.go

```diff
@@ -458,3 +463,484 @@ func TestScanHandlers_GetScanOperatingSystemVulnerabilitiesByID(t *testing.T) {
+					return []domain.ScanVulnerabilityDetail{}, nil
+				},
+				MockGetWebVulnerabilitiesForService: func(ctx context.Context, serviceID int32) ([]domain.WebVulnerability, error) {
+					return nil, errors.New("error getting web vuln for service")
+				},
+				MockGetServiceByID: func(ctx context.Context, serviceID int32) (*domain.Service, error) {
+					return &domain.Service{}, nil
+				},
+			},
+			scanScheduleService: &mock_services.MockScanScheduleService{},
+			hostService:         &mock_services.MockHostService{},
+			emailService:        &mock_services.MockEmailService{},
+			eventBus:            &events.NatsEventBus{},
+			r: func() *http.Request {
+				r := httptest.NewRequest("GET", "/api/scans/"+scanID.String()+"/services/"+strconv.Itoa(serviceID)+"/vulnerabilities", nil)
+				r.SetPathValue("id", scanID.String())
+				r.SetPathValue("service_id", strconv.Itoa(serviceID))
+				r = r.WithContext(context.WithValue(r.Context(), middleware.ContextRoles, []domain.Role{domain.RoleAdmin}))
+				return r
+			}(),
+			wantStatus:       http.StatusInternalServerError,
+			wantBodyContains: []string{},
+		},
+		{
+			name: "Empty for No Web and web Vulnerabilities → 200",
+			scanService: &mock_services.MockScanService{
+				MockGetScanByID: func(ctx context.Context, id uuid.UUID) (*domain.Scan, error) {
+					return &domain.Scan{Status: "Completed"}, nil
+				},
+				MockGetSeverityServiceCountsByScanAndServiceID: func(ctx context.Context, scanID uuid.UUID, serviceID int32) (tools.SeverityCounts, error) {
+					return tools.SeverityCounts{}, nil
+				},
+			},
+			vulnService: &mock_services.MockVulnerabilityService{
+				MockGetServiceVulnerabilityDetailByScanAndServiceID: func(ctx context.Context, scanID uuid.UUID, serviceID int32) ([]domain.ScanVulnerabilityDetail, error) {
+					return []domain.ScanVulnerabilityDetail{}, nil
+				},
+				MockGetWebVulnerabilitiesForService: func(ctx context.Context, serviceID int32) ([]domain.WebVulnerability, error) {
+					return []domain.WebVulnerability{}, nil
+				},
+				MockGetServiceByID: func(ctx context.Context, serviceID int32) (*domain.Service, error) {
+					return &domain.Service{}, nil
+				},
+			},
+			scanScheduleService: &mock_services.MockScanScheduleService{},
+			hostService:         &mock_services.MockHostService{},
+			emailService:        &mock_services.MockEmailService{},
+			eventBus:            &events.NatsEventBus{},
+			r: func() *http.Request {
+				r := httptest.NewRequest("GET", "/api/scans/"+scanID.String()+"/services/"+strconv.Itoa(serviceID)+"/vulnerabilities", nil)
+				r.SetPathValue("id", scanID.String())
+				r.SetPathValue("service_id", strconv.Itoa(serviceID))
+				r = r.WithContext(context.WithValue(r.Context(), middleware.ContextRoles, []domain.Role{domain.RoleAdmin}))
+				return r
+			}(),
+			wantStatus:       http.StatusOK,
+			wantBodyContains: []string{},
+		},
+		{
+			name: "Empty for No Web Vulnerabilities but not empty for web→ 200",
```

Comment:
`[Nit]` This name is convoluted and hard to read:


```suggestion
			name: "Service with only web vulnerabilities returns partial data → 200",
```